include/onnxruntime/core/graph/constants.h

```diff
@@ -37,49 +37,53 @@ constexpr const char* kMIGraphXExecutionProvider = "MIGraphXExecutionProvider";
 constexpr const char* kAclExecutionProvider = "ACLExecutionProvider";
 constexpr const char* kArmNNExecutionProvider = "ArmNNExecutionProvider";
 constexpr const char* kRocmExecutionProvider = "ROCMExecutionProvider";
-constexpr const char *providers_available[] = {
-  kCpuExecutionProvider,
+
+// EP used for internal testing
+constexpr const char* kInternalTestingExecutionProvider = "InternalTestingExecutionProvider";
```

Comment:
Can we exclude a test EP name from this file? This file is for all the officially supported EPs.

Reply:
I will move it to the internal framework/utils.h. It's used in ProviderIsCpuBased in framework/utils.cc which is used by partitioning logic, so it needs to exist at that level (or we need the EP API to have a property saying whether it's CPU based so we don't have hardcoded lists for that).

Reply:
Yeah, moving it to internal framework/utils.h is a better idea. Adding an interface to EPs raises more questions.